### ISSUE TITLE
[A11y] Collection Landing - Facets' "more >>" Modal

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1840]
### Operating System

Windows

### Browser

Edge

### Affected page

Collection Show

### Admin user?

Yes

### Issue Description

The modal which is displayed on "more >>" (within facets) is broken:

- You can shift-tab off the "X" and your focus goes somewhere I can't identify; it should wrap around to the last focusable element within the modal.  (This might be just the "X" itself, but it should simply stay there if that's the only focusable element)
- Tabbing past the "X" gets you to the browser's address bar, bookmarks, etc.  Tabbing should keep you within the modal.
- Closing the modal loses your keyboard focus.  It should return you to the place you were at when opening the modal.
- The modal's "X" button doesn't have any text readable by a screen reader.
- Opening the modal doesn't announce anything.  The dialog's role is probably incorrect.

### Suggested Fix

Basically the implementation has to follow the modal pattern precisely as defined in the W3C page: https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal

**Comment by jechols** [comment]
It's also worth noting I'm testing against a modal that simply says "Network Error".  Maybe the normal modal functions better, but even so the error modal still needs to be accessible.

**Comment by jechols** [comment]
QA requires a11y expertise to verify the modal functions as described in the w3c pattern

**Comment by jechols** [comment]
This isn't announcing that it's a dialog. I can't figure out why, though - the DOM seems like it should be working. I'm going to pass this since we need to focus on the bigger problems, but I suspect this is still a problem, even if not a true failure :-/
